### DISCUSSION TITLE
I don't know how works the Key

### DISCUSSION THREAD

**Discussion by R3B4T0** [lcobucci/jwt#754] [Q&A] [answered]
Error: "Cannot instantiate interface Lcobucci\JWT\Signer\Key"

Program: 

use Lcobucci\JWT\Signer\Key;

public function chat()
    {
        $username = $this->getUser()->getUsername();
        $token = (new Builder())
            ->withClaim('mercure', ['subscribe' => [sprintf("/%s", $username)]])
            ->getToken(
                new Sha256(),
                new Key($this->getParameter('mercure_secret_key'))
            )
        ;

        $response = $this->render('conversacion/index.html.twig', [
            'controller_name' => 'UsuarioController'
        ]);

        $response->headers->setCookie(
            new Cookie(
                'mercureAuthorization',
                $token,
                (new \DateTime())
                ->add(new \DateInterval('PT2H')),
                '/.well-known/mercure',
                null,
                false,
                true,
                false,
                'strict'
            )
        );

        return $response;
    }

**Comment by Ocramius** [comment, marked as answer]
That's because `Key` is an interface: 

https://github.com/lcobucci/jwt/blob/207955dfbce28b4fbdc12367fbccf9546c66eb2d/src/Signer/Key.php#L6

Are you reading the docs for the correct version of the library?

**Reply by R3B4T0** [reply]
When I implement Key, I get this error:

Error: Class App\Controller\UsuarioController contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (Lcobucci\JWT\Signer\Key::contents, Lcobucci\JWT\Signer\Key::passphrase)

**Reply by lcobucci** [reply]
We already provide some implementations for the Key, so you don't need to do that. Check https://lcobucci-jwt.readthedocs.io/en/latest/configuration/#configuration-initialisation for some ideas on how to instantiate them.

**Reply by Ocramius** [reply]
Hey @R3B4T0, there are already pre-existing implementations of `Key` in this library: https://github.com/lcobucci/jwt/tree/207955dfbce28b4fbdc12367fbccf9546c66eb2d/src/Signer/Key

As a general rule of thumb, if you do not understand how to implement the `Key` interface, you probably shouldn't do it on your own.

Please check https://github.com/lcobucci/jwt/blob/207955dfbce28b4fbdc12367fbccf9546c66eb2d/docs/configuration.md
